provider/disk_utils.py

```diff
@@ -0,0 +1,129 @@
+    """
+    if os_type == "windows":
+        idx_info = get_disk_props_by_serial_number(session, serial, ["Index"])
+        if idx_info:
+            return idx_info["Index"]
+    else:
+        return get_linux_drive_path(session, serial)
+
+
+def init_disk_by_id(
+    params,
+    vm,
+    did,
+    image_name,
+    dst_dir=None,
+    fstype=None,
+):
+    """
+    Initializes the disk in the guest based on the received device ID.
+    By initialize, it means obtain the reference of the disk (ID or path) and
+    later in Linux systems, create the filesystem + mounting or formatting the
+    disk in Windows ones.
+
+    :param params: Dictionary with the test parameters.
+    :param vm: The VM object.
+    :param did: The ID of the VM's disk.
+    :param image_name: The image name. e.g: stg0
+    :param dst_dir: The destination directory for mounting.
+    :param fstype: The kind of filesystem (ntfs or xfs).
+    """
+    session = vm.wait_for_login()
+    is_windows = params.get("os_type") == "windows"
+    image_params = params.object_params(image_name)
+    img_size = image_params.get("image_size")
+
+    if fstype is None:
+        fstype = "ntfs" if is_windows else "xfs"
+
+    if not dst_dir:
+        dst_dir = f"{did}:\\{image_name}" if is_windows else f"/mnt/{image_name}"
+        if is_windows:
+            session.cmd(f"mkdir {dst_dir}")
+        else:
+            session.cmd(f"mkdir -p {dst_dir}")
+
+    if is_windows:
+        utils_disk.update_windows_disk_attributes(session, did)
+        utils_disk.clean_partition_windows(session, did)
+        return utils_disk.configure_empty_disk(session, did, img_size, "windows")[0]
+    else:
+        # Send only the disk ID
+        disk_id = did.split("/")[-1]
+        utils_disk.create_filesyetem_linux(session, disk_id, fstype)
+        if utils_disk.mount(did, dst_dir, fstype, session=session):
+            return dst_dir
+
+
+def execute_io_test(
+    params,
+    vm,
+    image,
+    serial,
+    fstype=None,
+    dst_dir=None,
+    io_command=None,
+    clean=True,
+    ignore_all_errors=False,
+    timeout=60,
+):
+    """
+    Execute the I/O write test on the disk image inside the guest.
+    Note: The target disk image should be the uninitialized disk.
+
+    :param params: Dictionary with the test parameters.
+    :param vm: The VM object.
+    :param image: The image name. e.g: stg0
+    :param serial: The serial number of the disk
+    :param dst_dir: The destination directory for I/O write testing.
+    :param io_command: The command for I/O test.
+    :param clean: If True, cleans the environment
+    :param ignore_all_errors: Checks the errors of the I/O command
+    :param timeout: The timeout for the I/O command.
+    """
+    session = vm.wait_for_login()
+    is_windows = params["os_type"] == "windows"
+
+    did = get_disk_by_serial(params.get("os_type"), serial, session)
+    disk = init_disk_by_id(params, vm, did, image, dst_dir, fstype)
+
+    if not dst_dir:
```

Comment:
the disk indicates the final mount point , ignore handle dst_dir here

Reply:
@qingwangrh `dst_dir` is used in `is_mount` and `umount` if not initialized, these two functions will fail

Reply:
@mcasquer the dst_dir is updated to "disk" after init_disk_by_id , so you need to pass the "disk" to umount.
BYW,  you may rename it to "mount_point"  is easy to read.

Reply:
Got it, sorry for the misunderstood :)

Reply:
Anyways, code updated